src/lib.rs

```diff
@@ -27,6 +29,7 @@ mod superblock;
 mod util;
 
 use alloc::boxed::Box;
+use alloc::vec;
```

Comment:
nit: Not sure how this is related.

Reply:
Mentioned in the commit msg but forgot to include in the PR description:

> Also add an import of `alloc::vec` that is now needed when compiling without `std`.

Reason being, if `std` is enabled then a bunch of stuff is automatically in scope: https://doc.rust-lang.org/std/prelude/index.html